dashboard/src/components/leaflet/MapTooltip.tsx

```diff
@@ -0,0 +1,119 @@
+import { useMap } from 'react-leaflet/hooks';
+import { useEffect, useRef, useId } from 'react';
+import { CountryDisplay } from '../language/CountryDisplay';
+import { FlagIconProps } from '../icons';
+import { getCountryName } from '@/utils/countryCodes';
+import { cn } from '@/lib/utils';
+import { createPortal } from 'react-dom';
+import React from 'react';
+
+type MapTooltipProps = {
+  selectedCountry: {
+    code: string;
+    visitors: number;
+  } | null;
+  size?: 'sm' | 'lg';
+};
+
+type TooltipContentProps = {
+  code: string;
+  visitors: number;
+  size: 'sm' | 'lg';
+};
+
+export function getTooltipId(alpha2: string) {
+  return `lltooltip-${alpha2}`;
+}
```

Comment:
Quibble: `alpha2` in itself doesn't provide much information, consider renaming to `countryCode`